internal/stacks/stackruntime/internal/stackeval/component.go

```diff
@@ -205,12 +219,18 @@ func (c *Component) ResultValue(ctx context.Context, phase EvalPhase) cty.Value
 		// exact type constraints for its output values and so each instance of
 		// a component can potentially produce a different object type.
 
-		if _, exists := insts[addrs.WildcardKey]; exists || insts == nil {
-			// If we don't even know what instances we have then we can't
-			// predict anything about our result.
+		if unknown {
+			// We can't predict the result if we don't know what the instances
+			// are, so we'll return dynamic.
 			return cty.DynamicVal
 		}
 
+		if insts == nil {
+			// Then we errored during instance calculation, this should have
```

Comment:
There is no way for us to somehow signal back that this requires attention right? E.g. a diagnostic or sth similar

Reply:
The `cty.NilVal` would indicate that something went wrong. We'd see a panic now if anything ever tried to reference this. But this shouldn't ever happen. Something should have called `c.CheckInstances` earlier in the process (eg. down on line 318) and if we got an invalid value back at that point we'd have stopped execution. If we're requesting the result value at all, we should be confident that we have valid instances otherwise we'd have errored earlier.

Reply:
Ok, that makes sense 👍